node/src/syncer.rs

```diff
@@ -170,8 +171,12 @@ where
                     self.on_header_sub_message().await;
                     self.fetch_next_batch().await;
                 }
-                Some(cmd) = self.cmd_rx.recv() => {
-                    self.on_cmd(cmd).await;
+                cmd = self.cmd_rx.recv() => {
+                    if let Some(cmd) = cmd {
+                        self.on_cmd(cmd).await;
+                    } else {
+                        return;
+                    }
```

Comment:
I prefer [`CancellationToken`](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html) for graceful shutdown. We should use that in `Drop` impls etc. Also for each task. I have a separate issue for that (https://github.com/eigerco/celestia-node-rs/issues/70).

Reply:
Removed this closed channel detection and replaced it with `CancellationToken` and `DropGuard` in `Syncer`